sphinx_immaterial/apidoc/object_toc.py

```diff
@@ -182,7 +184,131 @@ def _collect(
+                field_name = cast(docutils.nodes.field_name, field[0])
+                if field_name["ids"]:
+                    continue
+                field_id = docutils.nodes.make_id(field_name.astext())
+                if obj_id:
+                    field_id = f"{obj_id}-{field_id}"
+                field_name["ids"].append(field_id)
+
+        return nodes
+
+    sphinx.directives.ObjectDescription.run = run
+
+
+_monkey_patch_object_description_to_include_fields_in_toc()
+
+
+@html_translator_mixin.override
+def depart_rubric(
+    self,
+    node: docutils.nodes.rubric,
+    super_func: html_translator_mixin.BaseVisitCallback[docutils.nodes.rubric],
+) -> None:
+    self.add_permalink_ref(node, _("Permalink to this headline"))
+    super_func(self, node)
+
+
+def _monkey_patch_object_description_to_include_rubrics_in_toc():
+    orig_run = sphinx.directives.ObjectDescription.run
+
+    def run(self: sphinx.directives.ObjectDescription) -> List[docutils.nodes.Node]:
+        nodes = orig_run(self)
+
+        options = object_description_options.get_object_description_options(
+            self.env, self.domain, self.objtype
+        )
+        if not options["include_rubrics_in_toc"]:
+            return nodes
+
+        obj_desc = nodes[-1]
+        obj_content = obj_desc[-1]
+        for child in obj_content:
+            if not isinstance(child, docutils.nodes.rubric):
+                continue
+            rubric = cast(docutils.nodes.rubric, child)
+            if rubric["ids"]:
+                continue
+            rubric_id = docutils.nodes.make_id(rubric.astext())
+            rubric["ids"].append(rubric_id)
+
+        return nodes
+
+    sphinx.directives.ObjectDescription.run = run
+
+
+_monkey_patch_object_description_to_include_rubrics_in_toc()
+
+
+@html_translator_mixin.override
+def depart_term(
+    self: html_translator_mixin.HTMLTranslatorMixin,
+    node: docutils.nodes.term,
+    super_func: html_translator_mixin.BaseVisitCallback[docutils.nodes.term],
+) -> None:
+    self.add_permalink_ref(node, _("Permalink to this definition"))
+    super_func(self, node)
+
+
 def setup(app: sphinx.application.Sphinx):
+    app.setup_extension("sphinx_immaterial.apidoc.object_description_options")
+    object_description_options.add_object_description_option(
+        app, "include_in_toc", type_constraint=bool, default=True
+    )
+    object_description_options.add_object_description_option(
+        app, "toc_icon_text", type_constraint=Optional[str], default=None
+    )
+    object_description_options.add_object_description_option(
+        app, "toc_icon_class", type_constraint=Optional[str], default=None
+    )
+    object_description_options.add_object_description_option(
+        app, "include_fields_in_toc", type_constraint=bool, default=True
+    )
+    object_description_options.add_object_description_option(
+        app, "include_rubrics_in_toc", type_constraint=bool, default=True
```

Comment:
Just an observation... `include_rubrics_in_toc` used to default to `False`. I'm fine with it being default `True`, but just making you aware in case that was unintentional.

Reply:
Thanks, that was not intended, fixed in #160.